src/Glennsl__BsBenchmarkJs.re

```diff
@@ -0,0 +1,971 @@
+  let setOnAbort: (eventHandler, suiteOptions) => suiteOptions = (name, opt) => { setOnAbort(opt, name); opt; };
+
+  /** [ setOnComplete(suiteOptions) ] mutates [ suiteOptions ] by setting its 'onComplete' property. */
+  let setOnComplete: (eventHandler, suiteOptions) => suiteOptions = (name, opt) => { setOnComplete(opt, name); opt; };
+  
+  /** [ setOnCycle(suiteOptions) ] mutates [ suiteOptions ] by setting its 'onCycle' property. */
+  let setOnCycle: (eventHandler, suiteOptions) => suiteOptions = (name, opt) => { setOnCycle(opt, name); opt; };
+  
+  /** [ setOnError(suiteOptions) ] mutates [ suiteOptions ] by setting its 'onError' property. */
+  let setOnError: (eventHandler, suiteOptions) => suiteOptions = (name, opt) => { setOnError(opt, name); opt; };
+  
+  /** [ setOnReset(suiteOptions) ] mutates [ suiteOptions ] by setting its 'onReset' property. */
+  let setOnReset: (eventHandler, suiteOptions) => suiteOptions = (name, opt) => { setOnReset(opt, name); opt; };
+  
+  /** [ setOnStart(suiteOptions) ] mutates [ suiteOptions ] by setting its 'onStart' property. */
+  let setOnStart: (eventHandler, suiteOptions) => suiteOptions = (name, opt) => { setOnStart(opt, name); opt; };
+
+};
+
+module Suite = {
+
+  type t = suite;
+
+  module Internal = {
+
+    [@bs.module "benchmark"] [@bs.scope "Benchmark"] [@bs.new] external make: string => t = "Suite";
+    [@bs.module "benchmark"] [@bs.scope "Benchmark"] [@bs.new] external makeWithOptions: (string, suiteOptions) => t = "Suite";
+    [@bs.send.pipe: t] external add: (string, testFn) => t = "add";
+    [@bs.send.pipe: t] external addWithOptions: (string, testFn, options) => t = "add";
+    [@bs.send.pipe: t] external run: t = "run";
+    [@bs.send.pipe: t] external runWithOptions: suiteOptions => t = "run";
+    [@bs.send.pipe: t] external clone: t = "clone";
+    [@bs.send.pipe: t] external cloneWithOptions: suiteOptions => t = "clone";
+    [@bs.send.pipe: t] external emit: string => t = "emit";
+    [@bs.send.pipe: t] external getListeners: array(eventHandler) = "listeners";
+    [@bs.send.pipe: t] external getListenersByEvent: string => array(eventHandler) = "listeners";
+    [@bs.send.pipe: t] external removeListener: (string, eventHandler) => t = "off";
+    [@bs.send.pipe: t] external removeListenersByEvent: string => t = "off";
+    [@bs.send.pipe: t] external removeAllListeners: t = "off";
+    [@bs.send.pipe: t] external on: (string, eventHandler) => t = "on";
+    let eventListToString: list(eventType) => string =
+      eventList =>
+        Belt.List.reduceU(eventList, "", (. str, evt) => {
+          str ++ eventTypeToString(evt) ++ " "
+        });
+    let addBenchmarkToSuite: (. t, benchmark) => t =
+      (. suite, bench) =>
+        addWithOptions(
+          Benchmark.getName(bench),
+          Benchmark.getFn(bench),
+          Benchmark.getOptions(bench),
+          suite,
+        );
```

Comment:
I think this was just because it is used internally, and we might as well avoid currying. Ultimately, I plan to generate a `.rei` file for this library, and remove all the `Internal` modules from the interface.

Reply:
This might be just me, but when I see something "unusual" I start wondering why it is. And the cognitive overhead of that makes me think there should be a significant benefit to it that I want to understand, which just adds to it. If there's not a significant benefit,, then the cost of going outside the "norm" seems to outweigh it.

A very minot nit in this case, but just in general I think "might as well" should be skewed towards the norm, rather than some unusual trick that might have a non-obvious benefit.

Reply:
That is a very good point. I will change that.